From 5d5c4fd10c00dca6c77643b9893645f2b7935c70 Mon Sep 17 00:00:00 2001
From: Raveendran Somu <raveendran.somu@cypress.com>
Date: Tue, 11 Sep 2018 19:06:06 -0500
Subject: [PATCH 071/154] non-upstream: Changes to improve USB Tx throughput.

The inbound buffer been duplicated and returned to
the upper layer to increase the througput.

Below the improvement observed in different traffic
UDP Rx	UDP Tx	TCP Rx	TCP Tx
237	138	161	71	Without Tx improvement
238	155	162	137	With Tx improvement



Signed-off-by: Raveendran Somu <raveendran.somu@cypress.com>
---
 drivers/net/wireless/broadcom/brcm80211/brcmfmac/usb.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/usb.c b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/usb.c
index 7b36875..e38c56d 100644
--- a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/usb.c
+++ b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/usb.c
@@ -621,6 +621,7 @@ static int brcmf_usb_tx(struct device *dev, struct sk_buff *skb)
 		goto fail;
 	}
 
+	skb_orphan(skb);
 	req->skb = skb;
 	req->devinfo = devinfo;
 	usb_fill_bulk_urb(req->urb, devinfo->usbdev, devinfo->tx_pipe,
-- 
1.9.1

